xbmc/cores/VideoPlayer/DVDSubtitles/DVDSubtitlesLibass.cpp

```diff
@@ -130,64 +145,265 @@ bool CDVDSubtitlesLibass::DecodeHeader(char* data, int size)
+    style->ScaleX = 1.0;
+    style->ScaleY = 1.0;
+    // Extra space between characters causes the underlined
+    // text line to become more discontinuous (test on LibAss 15.1)
+    style->Spacing = 0;
+
+    bool isFontBold =
+        (subStyle.fontStyle == FontStyle::BOLD || subStyle.fontStyle == FontStyle::BOLD_ITALIC);
+    bool isFontItalic =
+        (subStyle.fontStyle == FontStyle::ITALIC || subStyle.fontStyle == FontStyle::BOLD_ITALIC);
+    style->Bold = isFontBold * -1;
+    style->Italic = isFontItalic * -1;
+
+    // Compute the font color, depending on the opacity
+    UTILS::Color subColor = ConvColor(subStyle.fontColor, subStyle.fontOpacity);
+    // Set default subtitles color
+    style->PrimaryColour = subColor;
+    // Set SecondaryColour may be used to prevent an onscreen collision
+    style->SecondaryColour = subColor;
+
+    // Compute the outline color
+    UTILS::Color outlineColor = ConvColor(subStyle.outlineColor);
+
+    // Configure the effects
+    if (subStyle.borderStyle == BorderStyle::OUTLINE)
+    {
+      style->BorderStyle = BORDER_STYLE_OUTLINE;
+      style->OutlineColour = outlineColor;
+      style->BackColour = ConvColor(UTILS::COLOR::BLACK); // Shadow color
+      style->Outline = 1 * scale;
+      style->Shadow = 0;
+    }
+    else
+    {
+      // Compute the color to be used for the box, depending on the opacity
+      style->BorderStyle = BORDER_STYLE_OUTLINE_BOX;
+      style->OutlineColour = outlineColor;
+      style->BackColour = ConvColor(subStyle.backgroundColor, subStyle.backgroundOpacity);
+      style->Outline = 1 * scale;
+      style->Shadow = 4; // Space between the text and the box edges
+    }
+
+    // Set the margins (in pixel)
+    style->MarginL = 20 * scale;
+    style->MarginR = style->MarginL;
+    style->MarginV = 20 * scale; // Vertical (direction depends on alignment)
+  }
+
+  // Set the alignment
+  if (subStyle.alignment == FontAlignment::TOP_LEFT)
+    style->Alignment = VALIGN_TOP | HALIGN_LEFT;
+  else if (subStyle.alignment == FontAlignment::TOP_CENTER)
+    style->Alignment = VALIGN_TOP | HALIGN_CENTER;
+  else if (subStyle.alignment == FontAlignment::TOP_RIGHT)
+    style->Alignment = VALIGN_TOP | HALIGN_RIGHT;
+  else if (subStyle.alignment == FontAlignment::MIDDLE_LEFT)
+    style->Alignment = VALIGN_CENTER | HALIGN_LEFT;
+  else if (subStyle.alignment == FontAlignment::MIDDLE_CENTER)
+    style->Alignment = VALIGN_CENTER | HALIGN_CENTER;
+  else if (subStyle.alignment == FontAlignment::MIDDLE_RIGHT)
+    style->Alignment = VALIGN_CENTER | HALIGN_RIGHT;
+  else if (subStyle.alignment == FontAlignment::SUB_LEFT)
+    style->Alignment = VALIGN_SUB | HALIGN_LEFT;
+  else if (subStyle.alignment == FontAlignment::SUB_CENTER)
+    style->Alignment = VALIGN_SUB | HALIGN_CENTER;
+  else if (subStyle.alignment == FontAlignment::SUB_RIGHT)
+    style->Alignment = VALIGN_SUB | HALIGN_RIGHT;
+
+  if (subtitleType == NATIVE)
```

Comment:
Duplicated code

Reply:
you mean for "ConfigureAssOverride" ?

Reply:
Yes, it seems this is already called on the begining of the method

Reply:
done